executor/brie.go

```diff
@@ -117,42 +148,99 @@ func (bq *brieQueue) registerTask(
 		},
 	}
 
-	taskID := atomic.AddUint64(&bq.nextID, 1)
-	bq.tasks.Store(taskID, item)
-
-	return taskCtx, taskID
+	// use base64 encode to avoid SQL injection
+	sql := fmt.Sprintf(`INSERT INTO mysql.brie_tasks (
+				kind, origin_sql, queue_time, data_path, conn_id, status, progress, cancel
+			) VALUES ('%s', '%s', '%s', '%s', %d, '%s', %f, %d);`,
+		info.kind.String(),
+		base64.StdEncoding.EncodeToString([]byte(info.originSQL)),
+		info.queueTime.String(),
+		base64.StdEncoding.EncodeToString([]byte(info.storage)),
+		info.connID,
+		item.progress.cmd,
+		item.progress.GetFraction(),
+		0,
+	)
+	s.Lock()
+	defer s.Unlock()
+	_, err := s.Execute(ctx, sql)
+	if err != nil {
+		return nil, 0, item, err
+	}
+	rs, err := s.Execute(ctx, "SELECT LAST_INSERT_ID();")
```

Comment:
done in https://github.com/pingcap/tidb/pull/21195/commits/4b9a04365bd59713c8729d08bab7e02f6570b217 for above comemnts and this comment. 

`CreateSessionForBRIEFunc` is still needed, because lightning will need a fixed session for multiple SQLs to implemente a transaction. 🤔